lazy_dataset/core.py

```diff
@@ -2229,16 +2247,19 @@ def __init__(self, *input_datasets):
                                                f'one dataset.'
                                                f'\n{self.input_datasets}')
         assert all([len(dataset) > 0 for dataset in self.input_datasets])
+        dataset_lengths = [len(ds) for ds in input_datasets]
         self.order = sorted([
-            ((example_index + 1) / len(dataset), dataset_index, example_index)
-            for dataset_index, dataset in enumerate(input_datasets)
-            for example_index in range(len(dataset))
+            ((example_index + 1) / ds_len, dataset_index, example_index)
+            for dataset_index, ds_len in enumerate(dataset_lengths)
+            for example_index in range(ds_len)
         ])
 
     def copy(self, freeze=False):
-        return self.__class__(
-            *[ds.copy(freeze=freeze) for ds in self.input_datasets]
-        )
+        # Use __new__ to not deepcopy attributes
```

Comment:
`# Use __new__ to not deepcopy attributes` -> `# Use __new__ to avoid __init__ overhead`?